developer_docs/public_api.md

```diff
@@ -0,0 +1,408 @@
+# Public API
+
+⚠️: This is currently a proposal, not an API that exists.
+
+# Authentication
+
+- Access to the API is available via a Personal Access Token, linked to an existing editor user account
+- Created via a logged-in user’s Settings page
+- Tokens can be deleted via the Settings page
+- Tokens are presented to the API using basic auth
+- Tokens can also be sent via the `Authorization: Bearer {your_access_token}` HTTP header
+
+# API Access
+
+- API writes are rate-limited to X per second, per access token
+- Operations are transactional, e.g. if a file is somehow invalid, the sketch won’t be left partially uploaded
+- All requests send and receive `Content-Type: application/json` unless otherwise stated
+
+# Models
+
+The API accepts and returns the following model objects, as JSON.
+
+## Sketch
+
+| Name  | Type              | Description                                                                          |
+| ----- | ----------------- | ------------------------------------------------------------------------------------ |
+| name  | String            | The sketch’s title                                                                   |
+| files | DirectoryContents | The files and directories in this sketch. See `DirectoryContents` for the structure. |
+
+
+    {
+      "id": String, // opaque ID
+      "name: String,
+      "files": DirectoryContents
+    }
+
+### Validations
+
+- `files` must have exactly one top-level file with the `.html` extension
+    - We could create a default `index.html` if not provided?
```

Comment:
i like this idea! maybe it also makes sense to add the default css as well?

Reply:
Added to the document.